woocommerce/google-listings-and-ads
# (R2.1f) Google tag enhanced conversion

### Changes proposed in this Pull Request:



Related to #2216.

1. If the enhanced conversion is enabled add the `{ 'allow_enhanced_conversions': true }` to the global site tag config (see [this article](https://support.google.com/google-ads/answer/13258081?visit_id=638283250727735978-1247451496&rd=2#Add_a_code_snippet&zippy=%2Cconfigure-your-conversion-page-google-tag)).
2. When the enhanced conversion is disabled, remove the above config from global site tag.
3. When the enhanced conversion is enabled, add the hashed user data on order received page so that it can be sent to analytics. The formatting for hashed data can be found in [this article](https://support.google.com/google-ads/answer/13262500?hl=en#zippy=identify-and-define-your-enhanced-conversions-variables%2Cset-up-standard-automatic-enhanced-conversions%2Cenable-enhanced-conversions-in-google-tag-manager-and-create-custom-javascript-variable:~:text=164%20standard.-,For%20hash%3A,-Use%20hex%20SHA256).
4.  When the enhanced conversion is disabled, do not add user data on order received page.


### Screenshots:


See testing screenshots in [this QA report](https://github.com/woocommerce/google-listings-and-ads/pull/2258#issuecomment-1991859193).

### Detailed test instructions:


Steps (when enabled):

1. If the Enhanced Conversion setting is not already enabled, navigate to WP Dashboard > Marketing > Google Listing & Ads > Settings and enable it.
2. Add a product to the cart and proceed to complete the checkout process.
3. Verify that the order is successfully placed and the thank you page is displayed to the customer.
4. View the source code of the thank you page.

Expected Result:

1. The configuration 'allow_enhanced_conversions:true' should be present in the Global Site Tag within the source code of the thank you page.
2. User data should be present in encrypted form on the thank you page. As shown in the screenshot.

### Additional details:


### Changelog entry

> Update - Set the "allow_enhanced_conversions" flag to Google tag when Enhanced Conversions is enabled.
